src/fast-path.js

```diff
@@ -469,6 +469,16 @@ FPp.needsParens = function() {
       } else if (parent.type === "AssignmentExpression") {
         return false;
       }
+
+      const grandParent = this.getParentNode(1)
```

Comment:
Could you
 - put this at the top of the case
 - add `{}` around case so it's scoped to that case part and not the entire function. If we want to use it in another place, it's going to yell saying that the variable is already defined elsewhere.
 - Add the following block just after the one that handles "ForStatement" as they are related.

Thanks!